train/tcn_train.py

```diff
@@ -0,0 +1,564 @@
+            y = label.data.tolist()
+            y_hat = label_out.max(dim=1)[1].tolist()
+
+            for u in y:
+                y_tot.append(u)
+            for u in y_hat:
+                y_hat_tot.append(u)
+
+            for yi, y_hati in zip(y, y_hat):
+                conf_matrix[yi, y_hati] += 1
+
+    print('\n confusion matrix (not normalized)')
+    print('\n', conf_matrix)
+
+    '''
+    Plot confusion matrix using heatmap.
+
+        Args:
+            data (list of list): List of lists with confusion matrix data.
+            labels (list): Labels which will be plotted across x and y axis.
+            output_filename (str): Path to output file.
+    '''
+    seaborn.set(color_codes=True)
+    # plt.figure(4+run, figsize=(9, 6))
+    plt.figure(figsize=(9, 6))
+
+    plt.title(f"{parameters.experiment} Confusion Matrix")
+
+    seaborn.set(font_scale=1.4)
+    ax = seaborn.heatmap(conf_matrix, annot=True, cmap="YlGnBu", cbar_kws={'label': 'Scale'})
+
+    ax.set_xticklabels(classes[parameters.experiment])
+    ax.set_yticklabels(classes[parameters.experiment])
+
+    ax.set(ylabel="True Label", xlabel="Predicted Label")
+
+    train_name: str = get_training_name(parameters.experiment, parameters.excluded_modalities)
+    plt.savefig(f'{parameters.results_path}{train_name}_confusion_{datetime.date(datetime.now())}_run_{run}.png',
+                bbox_inches='tight',
+                dpi=300)
+
+    plt.close()
+
+    print(' for classification report -   y_tot  and y hat_tot are: ')
+    print('y_tot = ', y_tot)
+    print('y hat tot = ', y_hat_tot)
+    # https://towardsdatascience.com/multi-class-metrics-made-simple-part-ii-the-f1-score-ebe8b2c2ca1
+    print(classification_report(y_tot, y_hat_tot, target_names=list((classes[parameters.experiment])), digits=3))
+    # https://towardsdatascience.com/multi-class-metrics-made-simple-the-kappa-score-aka-cohens-kappa-coefficient-bdea137af09c
+    print('cohen_kappa_score  = ', cohen_kappa_score(y_tot, y_hat_tot))
+    # https://scikit-learn.org/stable/modules/generated/sklearn.metrics.matthews_corrcoef.html
+    print('mathews correlation coefficient  = ', matthews_corrcoef(y_tot, y_hat_tot))
+    class_report = 0
+
+    return conf_matrix, class_report
+
+
+def test_model(test_config: TestConfig):
+    test_loader = DataLoader(test_config.test_set, batch_size=test_config.batch_size, num_workers=0, shuffle=True)
+
+    test_config.feat_ext.eval()
+    test_config.label_cls.eval()
+    test_config.plant_cls.eval()
+
+    tot_correct = 0.
+    tot_label_loss = 0.
+    with torch.no_grad():
+        for batch in test_loader:
+            if torch.cuda.is_available():
+                torch.cuda.empty_cache()
+            lbls = batch['label'].to(test_config.device)
+
+            x = batch['inception'].copy()
+
+            # del x['label']
+            # del x['plant']
+
+            for key in x:
+                x[key] = x[key].to(test_config.device)
```

Comment:
- [ ] use comprehension
```suggestion
            x = {mod: values.to(test_config.device) for mod, values in x.items()}
```